cli/command/system/completion.go

```diff
@@ -0,0 +1,99 @@
+package system
+
+import (
+	"strings"
+
+	"github.com/docker/cli/cli/command/completion"
+	"github.com/spf13/cobra"
+)
+
+var (
+	eventFilters = []string{"container", "daemon", "event", "image", "label", "network", "node", "scope", "type", "volume"}
+	eventNames   = []string{
+		"attach",
+		"commit",
+		"connect",
+		"copy",
+		"create",
+		"delete",
+		"destroy",
+		"detach",
+		"die",
+		"disable",
+		"disconnect",
+		"enable",
+		"exec_create",
+		"exec_detach",
+		"exec_die",
+		"exec_start",
+		"export",
+		"health_status",
+		"import",
+		"install",
+		"kill",
+		"load",
+		"mount",
+		"oom",
+		"pause",
+		"pull",
+		"push",
+		"reload",
+		"remove",
+		"rename",
+		"resize",
+		"restart",
+		"save",
+		"start",
+		"stop",
+		"tag",
+		"top",
+		"unmount",
+		"unpause",
+		"untag",
+		"update",
+	}
+)
+var eventTypes = []string{"config", "container", "daemon", "image", "network", "node", "plugin", "secret", "service", "volume"}
+
+func completeFilters(dockerCLI completion.APIClientProvider) completion.ValidArgsFn {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		if strings.HasPrefix(toComplete, "container=") {
+			// the pure container name list should be pulled out from ContainerNames.
+			names, _ := completion.ContainerNames(dockerCLI, true)(cmd, args, toComplete)
+			return prefixWith("container=", names), cobra.ShellCompDirectiveDefault
+		}
```

Comment:
Nice!! I was wondering the other day if something like this would work. I wanted to experiment if completion would be possible for the "advanced" CSV flags, but didn't give it a try yet.